CMakeVOL.cmake

```diff
@@ -34,11 +34,31 @@ if (HDF5_VOL_ALLOW_EXTERNAL)
     message (FATAL_ERROR "HDF5_ALLOW_EXTERNAL_SUPPORT must be set to 'GIT' to allow building of external HDF5 VOL connectors")
   endif ()
 
+  set (HDF5_LIB_TARGETS "")
+  set (HDF5_HL_LIB_TARGETS "")
+
+  if (BUILD_STATIC_LIBS)
+    list(APPEND HDF5_LIB_TARGETS ${HDF5_LIB_TARGET})
+    list(APPEND HDF5_HL_LIB_TARGETS ${HDF5_HL_LIB_TARGET})
+  endif()
+
+  if (BUILD_SHARED_LIBS)
+    list(APPEND HDF5_LIB_TARGETS ${HDF5_LIBSH_TARGET})
+    list(APPEND HDF5_HL_LIB_TARGETS ${HDF5_HL_LIBSH_TARGET})
+  endif()
+
   # For compatibility, set some variables that projects would
   # typically look for after calling find_package(HDF5)
   set (HDF5_FOUND 1)
```

Comment:
This is where these vars should have a prefix like the plugins.

Reply:
`HDF5_FOUND`, `HDF5_LIBRARIES`, and `HDF5_INCLUDE_DIRS`, are specifically set to match the expected output of `find_package(HDF5)`, which [would set these same variables](https://cmake.org/cmake/help/latest/module/FindHDF5.html#result-variables)

Reply:
> This is where these vars should have a prefix like the plugins.

This code is specifically removing the `find_package(HDF5)` call in an external connector's CMake code and then setting the CMake variables that that call would have set if it had been made. Basically replacing the `find_package` functionality since HDF5 will get built at the time the connector does.

Reply:
Right. In plugins and others I do the same thing except with a prefix - of course it requires some control over the other package or I we do this with zlib/szip I overwrite the CMakeLists.txt file that gets imported from the repo. See the config/cmake dirs LIBEAC and SZIP.

Reply:
Here using a prefix would break the builds since we don't have any control over the external packages really.